lib/tokenizers/encoding/transformation.ex

```diff
@@ -0,0 +1,46 @@
+defmodule Tokenizers.Encoding.Transformation do
+  @moduledoc """
+  Module containing handy functions to build the transformations list.
+
+  This list is aplied to an encoding using `Tokenizers.Encoding.transform/2`.
+  """
+
+  @type t :: [
+          {:pad, {non_neg_integer(), Tokenizers.Encoding.padding_opts()}},
+          {:truncate, {non_neg_integer(), Tokenizers.Encoding.truncation_opts()}},
+          {:set_sequence_id, non_neg_integer()}
+        ]
+
+  @doc """
+  Generates the padding transformation.
+
+  Check `Tokenizers.Encoding.pad/3` for more information.
+  """
+  @spec pad(non_neg_integer(), Tokenizers.Encoding.padding_opts()) ::
+          {:pad, {non_neg_integer(), Tokenizers.Encoding.padding_opts()}}
```

Comment:
I moved option types directly to the relevant functions, so we no longer have that type. We can bring it back, but it's also fine to just say `keyword()` since we point to the user to `Encoding.pad/3` anyway :)

```suggestion
  @spec pad(non_neg_integer(), keyword()) :: {:pad, {non_neg_integer(), keyword()}}
```

Reply:
@jonatanklosko I'm not sure that having opts inside function and not as a type is good idea (at least here). Using `keyword()` removes the ElixirLS ability to autosuggest, while having full list of opts in every function is a duplication and can lead to inconsistency, when in the future commits one place will be updated and other - no. WDYT?

Reply:
Ah, I wasn't aware ElixirLS does that, I don't have strong opinion, but that's a fair argument. So in this case we can bring the type back to share it :)

Reply:
Yep, this is how it looks:

![image](https://github.com/elixir-nx/tokenizers/assets/3989623/073da962-506a-4099-8c73-a5132b40c1d6)

It take it from spec, so `keyword()` is less expressive.

Reply:
I'll take a look how to return types at least for these functions